ledger-tool/src/main.rs

```diff
@@ -980,7 +980,7 @@ fn assert_capitalization(bank: &Bank) {
 
 /// Get the AccessType required, based on `process_options`
 fn get_access_type(process_options: &ProcessOptions) -> AccessType {
-    if process_options.boot_from_local_state {
+    if process_options.use_snapshot_archives_at_startup == UseSnapshotArchivesAtStartup::Never {
```

Comment:
I think we should attempt to avoid `if enum == ENUM` in this sort of case. We are planning on adding an additional variant, it's not immediately clear we wouldn't want to have `PrimaryForMaintenance` for that variant.

Using a match in this sort of case is better imo, so we can't forget to update when adding the new variant.